deployment/data-streams/changeset/fee-manager/call_pay_link_deficit.go

```diff
@@ -0,0 +1,71 @@
+package fee_manager
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/ethereum/go-ethereum/common"
+	goEthTypes "github.com/ethereum/go-ethereum/core/types"
+
+	"github.com/smartcontractkit/chainlink/deployment"
+	"github.com/smartcontractkit/chainlink/deployment/data-streams/changeset"
+	"github.com/smartcontractkit/chainlink/deployment/data-streams/changeset/types"
+	"github.com/smartcontractkit/chainlink/deployment/data-streams/utils/mcmsutil"
+	"github.com/smartcontractkit/chainlink/deployment/data-streams/utils/txutil"
+	"github.com/smartcontractkit/chainlink/v2/core/gethwrappers/llo-feeds/generated/fee_manager_v0_5_0"
+)
+
+// PayLinkDeficitChangeset pay the LINK deficit for a given config digest
+var PayLinkDeficitChangeset deployment.ChangeSetV2[PayLinkDeficitConfig] = &payLinkDeficit{}
+
+type payLinkDeficit struct{}
+
+type PayLinkDeficitConfig struct {
+	ConfigPerChain map[uint64][]PayLinkDeficit
+	MCMSConfig     *changeset.MCMSConfig
+}
+
+type PayLinkDeficit struct {
+	FeeManagerAddress common.Address
+	ConfigDigest      [32]byte
+}
+
+func (a PayLinkDeficit) GetContractAddress() common.Address {
+	return a.FeeManagerAddress
+}
+
+func (cs payLinkDeficit) Apply(e deployment.Environment, cfg PayLinkDeficitConfig) (deployment.ChangesetOutput, error) {
+	txs, err := txutil.GetTxs(
+		e,
+		types.FeeManager.String(),
+		cfg.ConfigPerChain,
+		LoadFeeManagerState,
+		doPayLinkDeficit,
+	)
+	if err != nil {
+		return deployment.ChangesetOutput{}, fmt.Errorf("failed building Withdraw txs: %w", err)
+	}
+
+	return mcmsutil.ExecuteOrPropose(e, txs, cfg.MCMSConfig, "Withdraw proposal")
+}
+
+func (cs payLinkDeficit) VerifyPreconditions(e deployment.Environment, cfg PayLinkDeficitConfig) error {
```

Comment:
nit: We might want to choose better receiver names. I was a little surprised that we were shadowing the name `cs` a couple of lines below. Definitely not a big deal, ofc.

Reply:
Did not notice this - good eye. Theres a lot of places where this is happening at the moment unfortunately

Reply:
Not an issue, just surprised me. :)

Reply:
I believe the function approach for changeset is cleaner and don't need a receiver